SSM/Abilities/BoneExplosion.java

```diff
@@ -22,19 +25,49 @@ public BoneExplosion() {
     }
 
     public void activate() {
-        List<Entity> canHit = owner.getNearbyEntities(4, 4, 4);
-        canHit.remove(owner);
+        owner.getWorld().playSound(owner.getLocation(), Sound.ENTITY_SKELETON_DEATH, 10L, 1L);
+        ItemStack bone = new ItemStack(Material.BONE, 1);
+        for (int i = 0; i < boneAmount; i++) {
+            ItemMeta x = bone.getItemMeta();
+            x.setDisplayName("ooga booga"+i);
+            bone.setItemMeta(x);
+            Item firing = owner.getWorld().dropItem(owner.getEyeLocation(), bone);
+            BoneExplosion.boneProjectile projectile = new BoneExplosion.boneProjectile(plugin, owner, name, firing);
+            projectile.setOverridePosition(owner.getEyeLocation().subtract(0, -1, 0));
+            projectile.launchProjectile();
+            List<Entity> canHit = owner.getNearbyEntities(4, 4, 4);
+            canHit.remove(owner);
 
-        if (canHit.size() <= 0) {
-            return;
-        }
+            if (canHit.size() <= 0) {
+                return;
+            }
+
+            for (Entity entity : canHit) {
+                if ((entity instanceof LivingEntity)) {
+                    ((LivingEntity) entity).damage(6.0);
+                    Vector p = entity.getLocation().toVector();
+                    Vector a = owner.getLocation().toVector();
+                    Vector pre = a.subtract(p);
+                    Vector velocity = pre.normalize().multiply(1.8);
 
-        for (Entity entity : canHit) {
-            if ((entity instanceof LivingEntity)) {
-                ((LivingEntity) entity).damage(6.0);
-                entity.getWorld().playSound(entity.getLocation(), Sound.ENTITY_SKELETON_HURT, 10, 10);
+                    entity.setVelocity(new Vector(velocity.getX()*-.75, 0.4, velocity.getZ()*-.75));
```

Comment:
Don’t multiply in the new vector, multiply the velocity variable if its going to be the same thing. -3/4 * 1.8 or 9/5 is -27/20 or -1.35

Reply:
no